tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -483,6 +484,23 @@ void ordering_test_cases() {
 
         spaceship_test<std::strong_ordering>(c_mem[0], c_mem[0], c_mem[1]);
     }
+    { // slice
+        std::valarray<int> a(20);
+        for (int i = 0; i < 20; ++i) {
+            a[i] = i;
+        }
+
+        const auto a1 = a[std::slice(2, 3, 4)];
+        const auto a2 = a[std::slice(2, 3, 4)];
+        const auto a3 = a[std::slice(3, 3, 4)];
+        const auto a4 = a[std::slice(2, 4, 4)];
+        const auto a5 = a[std::slice(2, 3, 3)];
+
+        static_assert((a1 == a2) == true);
+        static_assert((a1 == a3) == false);
+        static_assert((a1 == a4) == false);
+        static_assert((a1 == a5) == false);
+    }

```

Comment:
The test is not testing `operator==(const slice&, const slice&)` as intended. `a1` .. `a5` are `slice_array<int>`, not `slice`. It is unexpectedly passing (after minor fixup) due to #1630.